x-pack/plugins/aiops/public/shared_lazy_components.tsx

```diff
@@ -23,8 +35,77 @@ const LazyWrapper: FC = ({ children }) => (
  * Lazy-wrapped ExplainLogRateSpikes React component
  * @param {ExplainLogRateSpikesProps}  props - properties specifying the data on which to run the analysis.
  */
-export const ExplainLogRateSpikes: FC<ExplainLogRateSpikesProps> = (props) => (
-  <LazyWrapper>
-    <ExplainLogRateSpikesLazy {...props} />
-  </LazyWrapper>
-);
+export const ExplainLogRateSpikes: FC<ExplainLogRateSpikesProps> = (props) => {
```

Comment:
We should try to keep this component as small as possible, otherwise it will grow the loading bundle, ideally this file should not change at all. You could for example create a new component `ExplainLogRateSpikesUrlStateWrapper` that gets consumed here as `ExplainLogRateSpikesUrlStateWrapperLazy`. This new component would then contain `ExplainLogRateSpikes`.

Reply:
Good call - updated in fd3cbb696ab1d3bc4f6c9cbcfd9f29124e986bf1

Reply:
Can the `UrlStateContextProvider` and related code (e.g. `setUrlState`) also be moved into `ExplainLogRateSpikesWrapperLazy`?

Reply:
Looking at the code again, it seems you moved it correctly to the inner component already 👍  but just missed removing it here?